dropshot/src/server.rs

```diff
@@ -73,6 +72,37 @@ pub struct DropshotState<C: ServerContext> {
     /// Worker for the handler_waitgroup associated with this server, allowing
     /// graceful shutdown to wait for all handlers to complete.
     pub(crate) handler_waitgroup_worker: DebugIgnore<waitgroup::Worker>,
+    /// Determines how to construct responses for server errors.
+    pub(crate) error_handler: Box<
+        dyn Fn(ErrorContext<'_, C>, ServerError) -> Response<Body>
+            + Send
+            + Sync,
+    >,
```

Comment:
I'm not sure if this is really the nicest interface for the "unhandled dropshot-internal error" API --- open to suggestions here!

Reply:
Another option I considered was having the user provide a _type_ when constructing the server, and bounding it with `IntoErrorResponse + From<ServerError>`, and having the `From` impl for that type construct the error. That way, we could also be aware of the schema for the global catchall error...but, it's trickier to provide request metadata that way if it's just a `From` impl. Potentially, we could instead have this handler return a type implementing `IntoErrorResponse`, but we'd then have to erase that type unless we wanted the server to be generic over  it..

Reply:
See also comments in https://github.com/oxidecomputer/dropshot/pull/1164#discussion_r1830222468